src/java/picard/analysis/directed/GcBiasMetricsCollector.java

```diff
@@ -0,0 +1,286 @@
+        this.windowSize = windowSize;
+        this.bisulfite = bisulfite;
+        this.gcByRef = gcByRef;
+        this.windowsByGc = windowsByGc;
+        setup(accumulationLevels, samRgRecords);
+    }
+    /////////////////////////////////////////////////////////////////////////////
+    // This method is called once Per samRecord
+    /////////////////////////////////////////////////////////////////////////////
+    @Override
+    protected GcBiasCollectorArgs makeArg(final SAMRecord rec, final ReferenceSequence ref) {
+        return new GcBiasCollectorArgs(rec, ref);
+    }
+
+    /////////////////////////////////////////////////////////////////////////////
+    //Make a GcBiasCollector with the given arguments
+    /////////////////////////////////////////////////////////////////////////////
+    @Override
+    protected PerUnitMetricCollector<GcBiasMetrics, Integer, GcBiasCollectorArgs> makeChildCollector(final String sample, final String library, final String readGroup) {
+        return new PerUnitGcBiasMetricsCollector(sample, library, readGroup);
+    }
+
+    @Override
+    public void acceptRecord(final SAMRecord rec, final ReferenceSequence ref) {super.acceptRecord(rec, ref);}
+
+    /////////////////////////////////////////////////////////////////////////////
+    //A collector for individual GcBiasMetrics for a given SAMPLE or SAMPLE/LIBRARY
+    //or SAMPLE/LIBRARY/READ_GROUP (depending on aggregation levels)
+    /////////////////////////////////////////////////////////////////////////////
+    public class PerUnitGcBiasMetricsCollector implements PerUnitMetricCollector<GcBiasMetrics, Integer, GcBiasCollectorArgs> {
+        Map<String, GcObject> gcData = new HashMap<String, GcObject>();
+        String sample = null;
+        String library = null;
+        String readGroup = null;
+
+        /////////////////////////////////////////////////////////////////////////////
+        //Records the accumulation level for each level of collection and initializes
+        // a GcObject for this accumulation level
+        /////////////////////////////////////////////////////////////////////////////
+        public PerUnitGcBiasMetricsCollector(final String sample, final String library, final String readGroup) {
+            this.sample = sample;
+            this.library = library;
+            this.readGroup = readGroup;
+            final String prefix;
+            if (this.readGroup != null) {
+                prefix = this.readGroup;
+                gcData.put(prefix, new GcObject());
+            } else if (this.library != null) {
+                prefix = this.library;
+                gcData.put(prefix, new GcObject());
+            } else if (this.sample != null) {
+                prefix = this.sample;
+                gcData.put(prefix, new GcObject());
+            } else {
+                prefix = "All_Reads";
```

Comment:
Is a symbol a variable type that I don't know of? I tried googling but am not sure I am understanding what you mean..